### ISSUE TITLE
[Android] Notes are no longer synced if they are deleted from the App while the Desktop version is closed

### ISSUE THREAD

**Issue by Softvision-RemusDranca** [mozilla/notes#1254]
**[Affected Platforms]:**
- Android 8.0.0 (Pixel One)
- Android 8.0.0 (Samsung S8)
- Android 6.0 (Huawei P8 Lite)
- All Windows

**[Prerequisites]:**
- Have the latest version of the "Firefox Notes" application (notes-1_0-qa-v2691) installed on the device.
- Have the latest version of the Notes extension installed on a clean new Firefox profile.
- Be logged in the app with the same account you used on the Desktop version.
- Have a few notes created.
- Both devices are synced.
- The browser is closed.

**[Steps to reproduce]:**
1. Open the App.
2. Select a note and delete it.
3. Open the Firefox browser with the profile prerequisites.
4. Click the Sync button from the bottom of the Sidebar.
5. Sync the App and observe the behavior.

**[Expected result]:**
- Notes are synced and the deleted note is redisplayed.

**[Actual result]:**
- The previously deleted note is not displayed.

**[Notes]:**
- New notes created in the browser still sync with the App.
- The deleted note is still not brought back after you reconnect in the App.
- All Notes sync correctly only after you reconnect on desktop.
- Attached is a screen recording of the issue:

![notes 3](https://user-images.githubusercontent.com/40061643/42278963-f545be84-7fa4-11e8-865b-a214f552c8e5.gif)

**Comment by vladikoff** [comment]
I cannot trigger this issue locally @Softvision-RemusDranca  :(

**Comment by vladikoff** [comment]
@Softvision-RemusDranca actually maybe I can....., seems like something to do with pressing sync from the Drawer

**Comment by vladikoff** [comment]
@Softvision-RemusDranca Could you please check if the same happens to you if you sync by "Pull to Refresh" instead of pressing "Sync" in the drawer?

**Comment by Softvision-RemusDranca** [comment]
Yes, I can reproduce this issue if I sync by "Pulling to Refresh".

**Comment by vladikoff** [comment]
@Softvision-RemusDranca so after you delete notes, we "Sync" automatically. Could you give it a chance to sync automatically and see if it still reproduces the issue?

I have a feeling because you manually sync in your "STR" it is why it doesn't properly delete the notes

**Comment by vladikoff** [comment]
Pushed a fix for this that should help fix up the sync races, let me know how it goes in testing

**Comment by Softvision-RemusDranca** [comment]
This issue is still reproducible with the latest version of "Firefox Notes" application (1.0.3 RC2) on Pixel (Android 8.0.0), Samsung S8 (Android 8.0.0) and Huawei P8 Lite (Android 6.0). 
I've tried the following sync methods: 
- Pressing the sync button from the drawer
- Pulling to "Refresh" 
- "Sync" automatically
Considering this I am reopening this issue.

**Comment by vladikoff** [comment]
Here's what I do:

1. Close the Notes WebExtension
2. Select multiple notes in mobile app and delete them
3. Reopen the sidebar in Notes WebExtension
4. Deleted notes are gone. 

To clarify in step "4." the notes are kept in your case?

**Comment by vladikoff** [comment]
@Softvision-RemusDranca Ah I see the problem, this is not an Android issue, but a WebExtension issue. If you have 2 WebExtension versions and delete a note in one, the other one will not get the change if Firefox was closed at the time.

**Comment by mzk1** [comment]
Is there any ETA on a fix for this? Really like Notes, so clean and simple, but this issue really bothers me.
Thanks!
